js/data.js

```diff
@@ -6,3 +6,33 @@ var data = {
   editing: null,
   nextEntryId: 1
 };
+var $img = document.getElementsByTagName('img')[0];
+var placeholder = $img.getAttribute('src');
+var $form = document.querySelector('form');
+var $elements = $form.elements;
+var $entryTitle = $elements.title;
+var $photoUrl = $elements.photoUrl;
+var $entryNotes = $elements.notes;
+
+$form.addEventListener('submit', handleSubmit);
+function handleSubmit(event) {
+  var $newEntry = {};
+  $newEntry[$entryTitle.name] = $entryTitle.value;
+  $newEntry[$photoUrl.name] = $photoUrl.value;
+  $newEntry[$entryNotes.name] = $entryNotes.value;
+  $newEntry.nextEntryId = data.nextEntryId;
```

Comment:
This property you are creating would be better named `entryId`. `nextEntryId` is appropriate in the data model, but this is just storing an id, not the next id.